### PR TITLE
Allowed empty data entry key for old invoke transactions.

### PR DESCRIPTION
**Pull request by nickeskov**

### CODE REVIEW

File: pkg/state/transaction_checker.go [hunk 1]
```diff
@@ -1202,7 +1202,7 @@ func (tc *transactionChecker) checkDataWithProofs(transaction proto.Transaction,
 		return nil, err
 	}
 	utf16KeyLen := tx.Version == 1 && !isRideV6Activated
-	if err := tx.Entries.Valid(utf16KeyLen); err != nil {
+	if err := tx.Entries.Valid(true, utf16KeyLen); err != nil {
```
Comment (esuwu) [inline pkg/state/transaction_checker.go:1205]: Shouldn't we also call here `IsProtobuf`? The logic would be the same, but there won't be any hard code

File: pkg/ride/runtime.go [hunk 2]
```diff
@@ -367,4 +367,5 @@ type environment interface {
 	rideV6Activated() bool
 	internalPaymentsValidationHeight() uint64
 	maxDataEntriesSize() int
+	protobufTransaction() bool
```
Comment (esuwu) [inline pkg/ride/runtime.go:370]: we'd better call this method `isProtobufTransaction`

**Review by esuwu — CHANGES_REQUESTED** [review]

File: pkg/state/transaction_checker.go (hunk 1 above)
Comment (nickeskov) [reply pkg/state/transaction_checker.go:1205]: Nope, for data transaction that's correct. Empty keys are forbidden for any data transaction version.

File: pkg/ride/runtime.go (hunk 2 above)
Comment (nickeskov) [reply pkg/ride/runtime.go:370]: Yep, I agree with you. But as you can see `EvaluationEnvironment` struct has `isProtobufTransaction` field. So, I can't call this method as you suggest.

Comment (esuwu) [reply pkg/ride/runtime.go:370]: isProtobufTx

**Review by esuwu — APPROVED** [review]

File: pkg/ride/runtime.go (hunk 2 above)
Comment (nickeskov) [reply pkg/ride/runtime.go]: Fixed

**Review by esuwu — APPROVED** [review]

**Review by esuwu — APPROVED** [review]

**Review by esuwu — APPROVED** [review]

**Pull request merged**